test/python/synthesis/test_siswap_synthesis.py

```diff
@@ -0,0 +1,137 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
```

Comment:
is this only because of the variable `u`? If yes maybe we could just rename that one and avoid a global disable 🙂

Reply:
yes. these variable naming rules are pretty annoying I think. I think we need to relax this at the qiskit level. Like why should naming my unitary `u` be a problem? @mtreinish what do you think?

Reply:
I'm in favour of relaxing the lint-enforced naming rules in general, although I've got to say the code in this PR isn't exactly the greatest advert for it.  There's a _lot_ of mix-and-match of variable styles with no clear relation to the mathematical paper they're drawn from (the non-iSWAP-specific code), and "why should naming my unitary `u` be a problem?" is indicative - `u` is mostly a bad name, because the majority of things we deal with in Qiskit are unitary.  In tests you can use some name that identifies it as the base of the test, and in synthesis code, I'm guessing there's _a lot_ of different unitaries kicking around at different points, so they ought to have proper names as well.

Reply:
But often the main point of a function is to do something with a single unitary. In this case naming it `unitary` seems reasonable. `XXDecomposer` does this and it looks pretty good there. I don't like `u`, but if I used a editor that had good support for Python semantics, I might not mind it. I have to distinguish occurrences of `u` visually. If it's really just as easy with a modern editor to deal with, I'd say go ahead and use it.